internal/cmd/controller/target/status_test.go

```diff
@@ -0,0 +1,504 @@
+						StagedDeploymentID: "off-id",
+					},
+					Status: fleet.BundleDeploymentStatus{
+						AppliedDeploymentID: "id",
+					},
+				},
+				DeploymentID: "id",
+			},
+			want: false,
+		},
+		{
+			name: "is not up-to-date if .Spec.DeploymentID does not match target.DeploymentID",
+			target: &Target{
+				Deployment: &fleet.BundleDeployment{
+					Spec: fleet.BundleDeploymentSpec{
+						DeploymentID:       "off-id",
+						StagedDeploymentID: "id",
+					},
+					Status: fleet.BundleDeploymentStatus{
+						AppliedDeploymentID: "id",
+					},
+				},
+				DeploymentID: "id",
+			},
+			want: false,
+		},
+		{
+			name: "is not up-to-date if .Status.AppliedDeploymentID does not match target.DeploymentID",
+			target: &Target{
+				Deployment: &fleet.BundleDeployment{
+					Spec: fleet.BundleDeploymentSpec{
+						DeploymentID:       "id",
+						StagedDeploymentID: "id",
+					},
+					Status: fleet.BundleDeploymentStatus{
+						AppliedDeploymentID: "off-id",
+					},
+				},
+				DeploymentID: "id",
+			},
+			want: false,
+		},
+		{
+			name: "is up-to-date",
+			target: &Target{
+				Deployment: &fleet.BundleDeployment{
+					Spec: fleet.BundleDeploymentSpec{
+						DeploymentID:       "id",
+						StagedDeploymentID: "id",
+					},
+					Status: fleet.BundleDeploymentStatus{
+						AppliedDeploymentID: "id",
+					},
+				},
+				DeploymentID: "id",
+			},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := upToDate(tt.target)
+			if got != tt.want {
+				t.Errorf("upToDate() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_updateStatusAndCheckUnavailable(t *testing.T) {
+	tests := []struct {
+		name    string
+		status  *fleet.PartitionStatus
+		targets []*Target
+		want    bool
+	}{
+		{
+			name: "should be available if all targets are available",
+			status: &fleet.PartitionStatus{
+				MaxUnavailable: 0,
+			},
+			targets: []*Target{
+				availableTarget(),
+				availableTarget(),
+			},
+			want: false,
+		},
+		{
+			name: "should be unavailable if one target is unavailable but 0 can be",
+			status: &fleet.PartitionStatus{
+				MaxUnavailable: 0,
+			},
+			targets: []*Target{
+				availableTarget(),
+				availableTarget(),
+				unavailableTargetMismatchedID(),
+			},
+			want: true,
+		},
+		{
+			name: "should be available if max unavailable is 1 and one target is unavailable",
+			status: &fleet.PartitionStatus{
+				MaxUnavailable: 1,
+			},
+			targets: []*Target{
+				availableTarget(),
+				availableTarget(),
+				unavailableTargetMismatchedID(),
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := updatePartitionStatus(tt.status, tt.targets)
+			if got != tt.want {
+				t.Errorf("updateStatusUnavailable() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestUnavailable(t *testing.T) {
+	tests := []struct {
+		name    string
+		targets []*Target
+		want    int
+	}{
+		{
```

Comment:
nit: this could be expanded with a couple of simple test cases: one with all targets available, and another one with all targets unavailable; that would prevent this test suite from still passing if we update `Unavailable` to return a hard-coded value of `2`, as unlikely as it seems ;)